### ISSUE TITLE
Question about the training process

### ISSUE THREAD

**Issue by JesseZZZZZ** [sunblaze-ucb/Intuitor#7]
![Image](https://github.com/user-attachments/assets/748b8b19-fff1-46da-b741-ea41298115e0)

Hi, dear authors. Thank you for your great work, we are following your work but encountered some problems. The picture above is the training process of the code (we made no changes based on your VeRL code). We are confused why the validation score and reward are descending...... Hoping for your reply !

**Comment by ArthurZhang02** [comment]
same question here. train batch size=128, epoch=2

**Comment by XuandongZhao** [comment]
Which prompt are you using? 

You may need to use this one:

instruction_following = "You are a helpful AI Assistant, designed to provided well-reasoned and detailed responses. You FIRST think about the reasoning process step by step and then provide the user with the answer. Please enclose your final answer in the box: \\boxed{}."

**Comment by JesseZZZZZ** [comment]
You are a helpful AI Assistant, designed to provided well-reasoned and detailed responses. You FIRST think about the reasoning process step by step and then provide the user with the answer. Please enclose your final answer in the box: \boxed{}.

Yeah, the prompt I am using is exactly you just said.

**Comment by JesseZZZZZ** [comment]
Are you using the base model? I guess it's because we are using the instruct model of Qwen2.5-3B...?



---- Replied Message ----
| From | Xuandong ***@***.***> |
| Date | 06/04/2025 13:20 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [sunblaze-ucb/Intuitor] Question about the training process (Issue #7) |
XuandongZhao left a comment (sunblaze-ucb/Intuitor#7)
image.png (view on web)

Here is one training run of verl with Qwen2.5-3B, but it was from many days ago and may have used different training configurations. We’ll check the current code and send you an update soon.

**Comment by XuandongZhao** [comment]
I am using the base model: `actor_rollout_ref.model.path=Qwen/Qwen2.5-3B` 

Why is your start point already 0.6?

**Comment by JesseZZZZZ** [comment]
Yeah, after I switched to the base model the training screen is almost the same as yours, thanks!

**Comment by XuandongZhao** [comment]
There could be several reasons for this, and here are some of our thoughts:

1. Instruct models may have already undergone extensive SFT and RL, achieving strong performance on the MATH dataset. Applying RLIF to further train on this single dataset could lead to model collapse—similar to what happens when a model is fine-tuned on a small dataset for too many epochs.

2. As demonstrated in DeepSeek-R1-Zero, which applies RL directly to a base model without using any SFT data, we are also interested in exploring similar approaches.

3. This doesn’t mean RLIF cannot be applied on top of instruct models. However, doing so may require a different dataset, careful hyperparameter tuning, and thoughtful prompt design.

**Comment by Clementine24** [comment]
Thank you for your response!

Regarding hyperparameter tuning, I note that you mentioned in paper that "This variant, denoted INTUITOR-Code in Table 1, modifies the setup as follows: the number of sampled completions per problem is increased to 14; the learning rate is reduced from 3 × 10−5 to 1 × 10−5; and the KL penalty is increased to β = 0.01." Is the RLIF more sensitive to the hyperparameters on the code generation task? why? and can you share how did you choose the hyperparameters?
